presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java

```diff
@@ -1070,43 +1072,89 @@ public void testViewAccessControl()
-                "CREATE VIEW test_invoker_view_access SECURITY INVOKER AS SELECT * FROM orders",
+                "CREATE VIEW test_invoker_view_column_access SECURITY INVOKER AS SELECT * FROM orders",
                 privilege("orders", CREATE_VIEW_WITH_SELECT_COLUMNS));
         assertAccessAllowed(
-                "SELECT * FROM test_invoker_view_access",
+                "SELECT * FROM test_invoker_view_column_access",
                 privilege(viewOwnerSession.getUser(), "orders", SELECT_COLUMN));
         assertAccessDenied(
-                "SELECT * FROM test_invoker_view_access",
+                "SELECT * FROM test_invoker_view_column_access",
                 "Cannot select from columns \\[.*\\] in table .*.orders.*",
                 privilege(getSession().getUser(), "orders", SELECT_COLUMN));
 
         // change access denied exception to view
-        assertAccessDenied("SHOW CREATE VIEW test_nested_view_access", "Cannot show create table for .*test_nested_view_access.*", privilege("test_nested_view_access", SHOW_CREATE_TABLE));
-        assertAccessAllowed("SHOW CREATE VIEW test_nested_view_access", privilege("test_denied_access_view", SHOW_CREATE_TABLE));
+        assertAccessDenied("SHOW CREATE VIEW test_nested_view_column_access", "Cannot show create table for .*test_nested_view_column_access.*", privilege("test_nested_view_column_access", SHOW_CREATE_TABLE));
+        assertAccessAllowed("SHOW CREATE VIEW test_nested_view_column_access", privilege("test_denied_access_view", SHOW_CREATE_TABLE));
 
-        assertAccessAllowed(nestedViewOwnerSession, "DROP VIEW test_nested_view_access");
-        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_view_access");
-        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_invoker_view_access");
+        assertAccessAllowed(nestedViewOwnerSession, "DROP VIEW test_nested_view_column_access");
+        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_view_column_access");
+        assertAccessAllowed(viewOwnerSession, "DROP VIEW test_invoker_view_column_access");
+    }
+
+    @Test
+    public void testViewFunctionAccessControl()
+    {
+        skipTestUnless(supportsViews());
+
+        Session viewOwnerSession = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.ofUser("test_view_access_owner"))
+                .setCatalog(getSession().getCatalog().get())
+                .setSchema(getSession().getSchema().get())
+                .build();
+
+        // TEST FUNCTION PRIVILEGES
+        // view creation permissions are only checked at query time, not at creation
+        assertAccessAllowed(
+                viewOwnerSession,
+                "CREATE VIEW test_view_function_access AS SELECT abs(1) AS c",
+                privilege("abs", GRANT_EXECUTE_FUNCTION));
+
+        assertAccessDenied(
+                "SELECT * FROM test_view_function_access",
+                "'test_view_access_owner' cannot grant 'abs' execution to user 'hive'",
```

Comment:
This is a bit misleading error. Do you have any suggestions?

Reply:
That's good enough. Eventually we should improve these messages by providing the whole chain of view references that lead to such errors.